File: sdks/typescript/src/subagent_stop_values.ts
/** Terminal Subagent hook status values projected without owning hooks. */
export const SubagentStopStatus = Object.freeze({
  Completed: "completed",
  Failed: "failed",
  Cancelled: "cancelled",
  TimedOut: "timed_out",
} as const);

export type SubagentStopStatus =
  (typeof SubagentStopStatus)[keyof typeof SubagentStopStatus];

export function subagentStopStatusAsStr(status: SubagentStopStatus): string {
  if (!Object.values(SubagentStopStatus).includes(status)) {
    throw new TypeError("invalid Subagent stop status");
  }
  return status;
}

File: sdks/python/src/echo_agent_sdk/subagent_stop_values.py
from __future__ import annotations

from enum import Enum


class SubagentStopStatus(str, Enum):
    COMPLETED = "completed"
    FAILED = "failed"
    CANCELLED = "cancelled"
    TIMED_OUT = "timed_out"

    def as_str(self) -> str:
        return self.value
